client/context/broadcast.go

```diff
@@ -131,6 +131,11 @@ func (ctx CLIContext) broadcastTxCommit(txBytes []byte) (*ctypes.ResultBroadcast
 		return res, err
 	}
 
+	if ctx.ResponseHandler != nil {
```

Comment:
I suppose this is a flexible way of doing this. May I suggest a few things?

* Rename `ResponseHandler` to `TxResponseHandler` to provide better context
* Take the two `if` statements below this and stuff them into a `DefaultTxResponseHandler`. This will DRY and cleanup `broadcastTxCommit`.
* Perhaps implement another `TxResponseHandler` that pretty prints the response (this may require merging  @sunnya97's PR).

Reply:
> * Rename `ResponseHandler` to `TxResponseHandler` to provide better context
> * Take the two `if` statements below this and stuff them into a `DefaultTxResponseHandler`. This will DRY and cleanup `broadcastTxCommit`.

Okay, pushed these two changes. I added `error` as the return to `TxResponseHandler` because `MarshalJSON` could return an error in `DefaultTxResponseHandler`. Looks like the error is handled higher up. I agree this is nice for DRY because it would allow someone to just wrap the default functionality.

> * Perhaps implement another `TxResponseHandler` that pretty prints the response (this may require merging  @sunnya97's PR).

Happy to help with that. Is it okay if it's in a separate PR connected with the discussion in #2953 ?